src/cluster_asm.c

```diff
@@ -7,8 +7,335 @@
  * GNU Affero General Public License v3 (AGPLv3).
  */
 
+#include "server.h"
 #include "cluster_asm.h"
+#include "cluster.h"
+#include "cluster_legacy.h"
 
-int dummy(void) {
-    return 0;
+#define ASM_IMPORT  (1 << 1)
+#define ASM_MIGRATE (1 << 2)
+
+typedef struct asmLink {
```

Comment:
why do we call it `asmLink`, it seems a slot-ranges migration task, right?

Reply:
you don't like the name? cluster_legacy.c uses "link" for the connection. I just added asm prefix similar to it. Do you prefer a better name?

Reply:
I think the connection just is a part of migration task, how about `asmTask`

Reply:
sure, changed to it